From 7afe5b51611afa983d87bc2f5aeb1c26971245f5 Mon Sep 17 00:00:00 2001
From: James Athappilly <jamesatha@meta.com>
Date: Thu, 2 Apr 2026 09:25:08 -0700
Subject: [PATCH 504/504] Fix const compiler error and format error

This fixes the build time error we are seeing on meta platforms
---
 lib/tomlc99/toml.c | 2 +-
 src/mctpd.c        | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/tomlc99/toml.c b/lib/tomlc99/toml.c
index e7b878e..2bd806f 100644
--- a/lib/tomlc99/toml.c
+++ b/lib/tomlc99/toml.c
@@ -2164,7 +2164,7 @@ int toml_rtod_ex(toml_raw_t src, double *ret_, char *buf, int buflen) {
   /* decimal point, if used, must be surrounded by at least one digit on each
    * side */
   {
-    char *dot = strchr(s, '.');
+    const char *dot = strchr(s, '.');
     if (dot) {
       if (dot == s || !isdigit(dot[-1]) || !isdigit(dot[1]))
         return -1;
diff --git a/src/mctpd.c b/src/mctpd.c
index 65bd65b..fade61f 100644
--- a/src/mctpd.c
+++ b/src/mctpd.c
@@ -5906,7 +5906,7 @@ static int parse_config_bus_owner(struct ctx *ctx, toml_table_t *bus_owner)
 	if (val.ok && val.u.i) {
 		uint64_t i = val.u.i;
 		if ((i > max_poll_interval_ms) || (i < min_poll_interval_ms)) {
-			warnx("endpoint polling interval invalid (%lu - %lu ms)",
+			warnx("endpoint polling interval invalid (%llu - %llu ms)",
 			      min_poll_interval_ms, max_poll_interval_ms);
 			return -1;
 		}
-- 
2.52.0

